docs/mockwbemserver.rst

```diff
@@ -867,13 +794,236 @@ This adds the instances to the repository display of the previous example:
     };
 
 
+.. _`Mocking multiple CIM namespaces`:
+
+Mocking multiple CIM namespaces
+-------------------------------
+
+Pywbem_mock allows creating multiple namespaces in the repository and installing
+elements in some or all of those namespaces.
+
+There is a default-namespace created when
+:class:`pywbem_mock.FakedWBEMConnection` is created from either the
+`default_namespace` of the constructor or with the system default of
+`root/cimv2`
+
+However, a working WBEM environment normally includes at least two namespaces
+
+1. An :term:`interop namespace` which contains the parts of the model that deal
+   with the WBEM server itself (ex. CIM_Namespace, CIM_ObjectManager, etc.) and
+   which must be publically discoverable without the user knowing what CIM
+   namespaces exist in the WBEM server The DM
+
+2. A user namespace where the CIM classes for the user model
+
+Pywbem_mock includes a user-defined provider for the CIM_Namespace class that
+can be enabled by adding code similar to the following to the setup of a
+mock environment.
+
+.. code-block:: text
+
+    # install the CIM_Namespace class either by directly compiling the class
+    # or using the DMTFCIMSchema class to compile the class
+
+    provider = CIMNamespaceProvider(self.cimrepository)
+    interop_namespace = "interop"   # or root/interop
+    conn.add_namespace(interop_namespace)
+    conn.register_provider, namespaces=interop_namespace)
+
 
 .. _`User-defined providers`:
 
 User-defined providers
 ----------------------
 
-TODO: Write this section. It should include defining them and registering them.
+Within the fake  WBEM server the response to client requests is normally
+determined by standard server responder methods that use data in the CIM
+repository to generate responses (ex. ``GetInstance`` gets the instances from the
+repository, possibly filters properties and attributes and returns the
+instance). However, it is expected that the user may want specific classes to
+generate responses that have side effects, or manage the returns differently
+than the normal responses.  Thus, for example, the DMTF defined CIM_Namespace
+class returns instances of each of the namespaces in the CIM repository. That means
+that somewhere an instance the class must be created each time a namespace is
+created or the responder must get the set of namespaces from the CIM repository
+to generate responses to ``GetInstance``, ``EnumerateInstance``, etc.
+
+Also, the :meth:`pywbem_mock.FakedWBEMConnection.InvokeMethod` client operation
```

Comment:
Unfortunately, the `:meth:` reference does not resolve inherited methods, and `InvokeMethod` is only on `WBEMConnection`, so this reference does not become a link.

Suggestion:
```
Also, the :meth:`~pywbem.WBEMConnection.InvokeMethod` client operation
```

Reply:
DONE